### PR TITLE
[Feature] Support to synchronize mysql entire databases 

### PR DESCRIPTION
**Pull request by chenhaifengkeda**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
- [ ] I have added documentation for my new feature or new function

### CODE REVIEW

**Comment by banmoy** [comment]
@chenhaifengkeda Have you tried [flink cdc 3.0](https://ververica.github.io/flink-cdc-connectors/master/content/%E5%BF%AB%E9%80%9F%E4%B8%8A%E6%89%8B/mysql-starrocks-pipeline-tutorial-zh.html) which supports to synchronize data and schema change

File: src/main/java/com/starrocks/connector/flink/catalog/starrocks/TableSchema.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+// specific language governing permissions and limitations
+// under the License.
+package com.starrocks.connector.flink.catalog.starrocks;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class TableSchema {
```
Comment (banmoy) [inline src/main/java/com/starrocks/connector/flink/catalog/starrocks/TableSchema.java:24]: You can reuse the existing schema framework. Please see StarRocksTable

File: src/main/java/com/starrocks/connector/flink/table/sink/StarRocksSinkOptions.java [hunk 2]
```diff
@@ -61,6 +61,7 @@ public enum StreamLoadFormat {
     private static final String FORMAT_KEY = "format";
 
     // required sink configurations
+    public static final ConfigOption<String> FENODES = ConfigOptions.key("fenodes").stringType().noDefaultValue().withDescription("starrocks fe http address.");
```
Comment (banmoy) [inline src/main/java/com/starrocks/connector/flink/table/sink/StarRocksSinkOptions.java:64]: duplicate with `load-url`

File: src/main/java/com/starrocks/connector/flink/connection/StarRocksJdbcConnectionOptions.java [hunk 3]
```diff
@@ -38,12 +38,24 @@ public class StarRocksJdbcConnectionOptions  implements Serializable {
     @Nullable
     protected final String password;
 
+    protected final String fenodes;
```
Comment (banmoy) [inline src/main/java/com/starrocks/connector/flink/connection/StarRocksJdbcConnectionOptions.java:41]: `fenodes` is not used commonly, and do not add it here

**Comment by banmoy** [comment]
Thanks for your contribution @chenhaifengkeda. Left some comments, and some other suggestions
1. add documentation to describe what this feature can do, and how to use it
2. add end-to-end tests to verify it can work
3. make DCO check pass

File: src/main/java/com/starrocks/connector/flink/catalog/starrocks/TableSchema.java (hunk 1 above)
Comment (chenhaifengkeda) [reply src/main/java/com/starrocks/connector/flink/catalog/starrocks/TableSchema.java]: get. I reuse existing starrocks schema framework

Comment (chenhaifengkeda) [reply src/main/java/com/starrocks/connector/flink/catalog/starrocks/TableSchema.java]: good idea

File: src/main/java/com/starrocks/connector/flink/catalog/StarRocksCatalog.java [hunk 4]
```diff
@@ -234,6 +238,63 @@ public Optional<StarRocksTable> getTable(String databaseName, String tableName)
...
+            throw new CatalogException(
+                    String.format("Failed to execute sql: %s", sql), e);
+        }
+    }
+
+    /**
+     * according to mysql schema, create a table
+     * @param schema
+     */
+    public void createTable(TableSchema schema) {
```
[the remark this answers is not in the document]
Comment (chenhaifengkeda) [reply src/main/java/com/starrocks/connector/flink/catalog/StarRocksCatalog.java]: get. I reuse StarRocksCatalog. createTable

File: src/main/java/com/starrocks/connector/flink/table/sink/StarRocksSinkOptions.java [hunk 5]
```diff
@@ -86,7 +87,7 @@ public enum StreamLoadFormat {
             .intType().defaultValue(30000).withDescription("Timeout in millisecond for connecting to the `load-url`.");
 
     public static final ConfigOption<Integer> SINK_SOCKET_TIMEOUT = ConfigOptions.key("sink.socket.timeout-ms")
-            .intType().defaultValue(-1).withDescription("Timeout in milliseconds that the http client waits for data or, put differently, " +
+            .intType().defaultValue(60000).withDescription("Timeout in milliseconds that the http client waits for data or, put differently, " +
```
[the remark this answers is not in the document]
Comment (chenhaifengkeda) [reply src/main/java/com/starrocks/connector/flink/table/sink/StarRocksSinkOptions.java]: the reason that happend err when submit run flink job.

File: src/main/java/com/starrocks/connector/flink/table/sink/StarRocksSinkOptions.java (hunk 2 above)
Comment (chenhaifengkeda) [reply src/main/java/com/starrocks/connector/flink/table/sink/StarRocksSinkOptions.java]: reuse load-url

File: src/main/java/com/starrocks/connector/flink/connection/StarRocksOptions.java [hunk 6]
```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 1999-2022 Alibaba Group Holding Ltd.
```
[the remark this answers is not in the document]
Comment (chenhaifengkeda) [reply src/main/java/com/starrocks/connector/flink/connection/StarRocksOptions.java]: haha

File: src/main/java/com/starrocks/connector/flink/connection/StarRocksOptions.java [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+ */
+package com.starrocks.connector.flink.connection;
+
+import com.starrocks.connector.flink.tools.IOUtils;
+
+import java.util.Properties;
+
+import static org.apache.flink.util.Preconditions.checkNotNull;
+
+public class StarRocksOptions extends StarRocksJdbcConnectionOptions {
```
[the remark this answers is not in the document]
Comment (chenhaifengkeda) [reply src/main/java/com/starrocks/connector/flink/connection/StarRocksOptions.java]: Move StarRocksOptions to package cdc

File: src/main/java/com/starrocks/connector/flink/connection/StarRocksJdbcConnectionOptions.java (hunk 3 above)
Comment (chenhaifengkeda) [reply src/main/java/com/starrocks/connector/flink/connection/StarRocksJdbcConnectionOptions.java]: already delete it

File: src/main/java/com/starrocks/connector/flink/catalog/StarRocksCatalog.java [hunk 8]
```diff
@@ -234,6 +238,63 @@ public Optional<StarRocksTable> getTable(String databaseName, String tableName)
...
+        List<String> tableList = executeSingleColumnStatement(
+                "SELECT TABLE_NAME FROM information_schema.`TABLES` " +
+                        "WHERE TABLE_SCHEMA=? and TABLE_NAME=?",
+                database,
+                table
+        );
+        return !tableList.isEmpty();
+    }
+
+    private List<String> executeSingleColumnStatement(String sql, Object... params) {
```
[the remark this answers is not in the document]
Comment (chenhaifengkeda) [reply src/main/java/com/starrocks/connector/flink/catalog/StarRocksCatalog.java:251]: delete it

**Comment by chenhaifengkeda** [comment]
1. add end-to-end test
2. add doc

**Comment by banmoy** [comment]
LGTM

**Review by banmoy — APPROVED** [review]

**Pull request merged**